build-tests/x86/rawhide/test-image-systemd-boot/appliance.kiwi

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<image schemaversion="7.5" name="kiwi-test-image-systemd-boot">
+    <description type="system">
+        <author>Marcus Schaefer</author>
+        <contact>marcus.schaefer@gmail.com</contact>
+        <specification>Fedora Rawhide Appliance</specification>
+    </description>
+    <preferences>
+        <version>2.0.0</version>
+        <packagemanager>dnf</packagemanager>
+        <bootsplash-theme>charge</bootsplash-theme>
+        <bootloader-theme>breeze</bootloader-theme>
+        <locale>en_US</locale>
+        <keytable>us</keytable>
+        <timezone>UTC</timezone>
+        <rpm-check-signatures>false</rpm-check-signatures>
+    </preferences>
+    <preferences>
+        <type image="oem" filesystem="xfs" kernelcmdline="console=ttyS0" firmware="efi" format="qcow2" efipartsize="200">
+            <oemconfig>
+                <oem-resize>false</oem-resize>
+            </oemconfig>
+            <bootloader name="systemd_boot" timeout="10"/>
+        </type>
+    </preferences>
+    <users>
+        <user password="$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0" home="/root" name="root" groups="root"/>
+    </users>
+    <repository type="rpm-md" alias="kiwi-next-generation" priority="1">
+        <source path="obs://Virtualization:Appliances:Staging/Fedora_Rawhide"/>
+    </repository>
+    <repository type="rpm-md" alias="Fedora_Rawhide">
+        <source path="obs://Fedora:Rawhide/standard"/>
+    </repository>
+<!--
+    <repository type="rpm-md" alias="Fedora_EPEL">
+        <source path="obs://Fedora:EPEL:8/standard"/>
+    </repository>
+-->
+<!--
+    <repository type="rpm-md" alias="Fedora38" sourcetype="metalink">
+        <source path="https://mirrors.fedoraproject.org/metalink?repo=fedora-38&amp;arch=x86_64"/>
+    </repository>
+    <repository type="rpm-md" alias="Fedora38-updates" sourcetype="metalink">
+        <source path="https://mirrors.fedoraproject.org/metalink?repo=updates-released-f38&amp;arch=x86_64"/>
+    </repository>
+-->
+    <packages type="image">
+        <!-- without grubby there are no kernels in /boot for some reason -->
+        <package name="grubby"/>
```

Comment:
We don't need this.

Reply:
Problem is if I don't install grubby, for some reason, all kernel files vanishes from /boot

Reply:
But if they're in `/boot`, sd-boot can't load them either, no? They need to be in the ESP.

Reply:
what I meant is this. Usually the /boot partition on Fedora is similar to other /boot locations. On my tests it looks like this
```
/boot/System.map-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64
/boot/config-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64
/boot/efi
/boot/initramfs-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64.img
/boot/symvers-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64.gz -> /lib/modules/6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64/symvers.gz
/boot/vmlinuz-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64
```
according to the package database the kernel (vmlinuz) is provided by ```kernel-core```. However, if you don't install ```grubby``` and not other component of grub the ```/boot``` partition does not contain the kernel anymore. I did not investigate why this is the case and just added grubby to the package list plus the comment

> But if they're in /boot, sd-boot can't load them either, no? They need to be in the ESP.

since kiwi checks if there is any kernel /boot we don't reach the point when sd-boot tooling comes into play

Reply:
So we have a broken assumption in kiwi then? Because when we're using sd-boot, we need a flag for whether we're using `/boot` or `/boot/efi` for that. To be clear, we can use `/boot` just fine if we install an EFI filesystem driver for the `/boot` partition. Fedora has the `efifs` package which includes various filesystem EFI drivers that we could install.

Reply:
sorry I can't follow you. If fedora has a package called ```kernel-core``` and it provides ```/boot/vmlinuz-xxxx``` then I expect this file to be there when I install this package. This is not the case here. There is some magic that runs in package scripts or whatever I don't know but the kernel is definitely missing in /boot if I don't install grubby and that feels pretty much like a bug. If kiwi would not run the kernel lookup check before bootctl or kernel-install the process would fail when these tools are called.

I do not understand how this is related to EFI or bootloader at all... there is no kernel anymore if I don't install grubby

Reply:
How are we doing the kernel lookup? Because kernel-core only provides those paths as ghost files...

I think I don't understand what's going on, maybe @DaanDeMeyer or @martinezjavier might understand this better.

Reply:
> Because kernel-core only provides those paths as ghost files

interesting. So what creates ```/boot/vmlinuz-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64``` and where is its origin ? If ```/boot``` cannot be trusted to contain a kernel I agree with you then the kiwi kernel lookup needs to be adapted to look elsewhere, wherever that is ? :)

Reply:
Also it looks like this is now going beyond this PR, if we need to lookup kernels in other places on Fedora that sounds like a different issue. I would also like to understand why under certain circumstances there are kernels in /boot and in other not. It feels very odd to me at the moment

Reply:
to what package belongs the kernel in Fedora, it's really strange. ```kernel``` contains nothing, ```kernel-core``` is a ghost, so what ??

Reply:
`kernel` is a metapackage to pull in everything else.

Reply:
In mkosi, when generating a bootable fedora image, we install the following packages:

- systemd
- util-linux
- dnf
- kernel-core
- kernel-modules
- dracut
- systemd-udev

That should be sufficient to get kernel-install to pick up the fedora kernels

Reply:
@DaanDeMeyer yep also using these packages. The issue I'm facing is that ```/boot``` does not provide the kernel. I was confused about the following:

```
rpm -ql kernel-core | grep vmlinuz
/boot/.vmlinuz-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64.hmac
/boot/vmlinuz-6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64
/lib/modules/6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64/.vmlinuz.hmac
/lib/modules/6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64/vmlinuz
```

From here I thought ok there is still the kernel in /boot. But as @Conan-Kudo mentioned that's just a ```%ghost```
so it does not really exist. And that leads in kiwi to

```
[ ERROR   ]: 10:45:34 | KiwiKernelLookupError: No kernel found in ...
```

If you install any package related to grub e.g ```grubby``` this problem "resolves" because for some reason
now kernel files are also in ```/boot```

As I like to support systemd-boot without any grub components I was thinking where is the kernel now ?

So I assume Fedora packages the kernel in

```
/lib/modules/6.1.0-0.rc4.20221111git4bbf3422df78.39.fc38.x86_64/vmlinuz
```

Is this correct ?

If so I need to add another search path in the kiwi code that looks up kernels

Can you confirm/advise ?

Thanks much

Reply:
Ah, yes, if you're running `kernel-install` manually, then you'll need to look up kernels in /usr/lib/modules. Same goes for Arch Linux. Gentoo puts them in a weird location. Debian/Ubuntu put them in /boot.

Reply:
yep I finally understood the concept, now it's clear. I pushed a commit here to extend the kernel lookup of kiwi to
* /boot
* /lib/modules
* /usr/lib/modules

With this change the integration builds and works without any grub components

Thanks